sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -443,12 +446,28 @@ void CAFMaker::beginRun(art::Run& run) {
 void CAFMaker::beginSubRun(art::SubRun& sr) {
   // get the POT
   // get POT information
-  art::Handle<sumdata::POTSummary> pot_handle;
-  sr.getByLabel("generator", pot_handle);
-
-  if (pot_handle.isValid()) {
-    fSubRunPOT = pot_handle->totgoodpot;
-    fTotalPOT += fSubRunPOT;
+  if(fIsRealData)
```

Comment:
We don't know this yet.

Reply:
I suggest a duck-typing work-around (which still sucks because it may be NuMI instead of BNB etc.):
```suggestion
  auto const bnb_spill = sr.getHandle<std::vector<sbn::BNBSpillInfo>>(fParams.BNBPOTDataLabel(), bnb_spill);
  fIsRealData = bnb_spill.isValid();
  if(fIsRealData)
```
(of course then you don't need to retrieve `bnb_spill` again in the `if` branch).

Unfortunately, for some mysterious reason `art::SubRun::isRealData()` is not exposed (thus, I have no idea if it is correctly set, either).

Reply:
There's a PR that Chris mentioned #209 that puts this in the fcl so I'm inclined to let it be set there? If not we'll probably want to coordinate that. I'm not sure where else this fIsRealData value may be used in the future.

Reply:
(commented elsewhere in this PR just a moment ago that this is not the case any more after discussion at yesterdays SBN Ana Framework meeting. Note I'm not really following this PR that much but see my PR getting mentioned so wanted to clarify)

Reply:
Never mind, apparently this is not the case anymore (wow this stuff is hard to follow sometimes...). This is likely a good way to do this in light of `fIsRealData` being set in the process step so I need to set it in the subRun elsewhere to make this work for POT accounting...everything else should be self-consistent even with that change

Reply:
Actually in that pull request I was suggesting that `fIsRealData` be _removed_, exactly to avoid bugs like this.
The workaround I suggest here may affect that too:
1. we can decide to leave `fIsRealData` data member be, but then the use in `beginRun()` will suffer the same issue as here
2. we can decide to drop it, and then `fIsRealData` here becomes a local `isRealData` variable.

Reply:
I haven't fully looked at the code (or indeed most of the comments in this PR review outside of the few items I've been following that involved my PR 😅 ), but just throwing in my 2 cents with my current thoughts:

1) `fIsRealData` doesn't seem be being used in `beginRun()` at the moment though? Is that going to change with other suggestions or just a general comment? Unless you (@PetrilloAtWork ) are maybe considering how one might use such a data member consistently throughout the module (e.g. later when I have things in the `produce` step) or something?

2) Why not just get rid of a variable altogether (unless the answer to point 1 demands it) and just do `if ( bnb_spill.isValid() )`?

3) Could you then also do something similar for NuMI and do `if ( bnb_spill.isValid() || numi_spill.isValid() ) `

Reply:
> fIsRealData doesn't seem be being used in beginRun() at the moment though? Is that going to change with other suggestions or just a general comment?

General comment. Until one week ago, `fIsRealData` was not used even in `beginSubRun()` and we were happy.
But then we forget the quirks, and six months from now we start using it, and much later we realize that we are missing information for one file per job (the issue is only for the first call of `begin[Sub]Run()`, after which `analyze()` kicks in; and those methods are in our situation called on each new input file).

Reply:
> Unfortunately, for some mysterious reason art::SubRun::isRealData() is not exposed

I made an art issue about this years ago, and it was closed WONTFIX. They say that in principle you could mix data and MC events into a single file. Not sure how this is consistent with including it on Event given the possibility of overlays...